src/components/map.tsx

```diff
@@ -1,71 +1,50 @@
-import styles from './map.module.css'
 import {
   useEffect,
   useCallback,
   useLayoutEffect,
   useRef,
-  useState,
+  useState
 } from 'react'
-import { addPoints, diffPoints, scalePoint } from '../helpers/math'
+import styled from 'styled-components'
 
-import { colours, pixelSize } from './layout'
+import { Colour, Point, Position, Size } from 'src/components/Types'
+import { pixelSize } from 'src/components/layout'
+import { generateTiles } from 'src/helpers/GenerateTiles'
+import { addPoints, diffPoints, scalePoint } from 'src/helpers/math'
 
-const mapSize = { width: 1000, height: 1000 }
+const mapSize: Size = { width: 1000, height: 1000 }
 
-type Point = {
-  x: number
-  y: number
-}
 const ORIGIN = Object.freeze({ x: 0, y: 0 })
-
+const ORIGIN_SIZE = Object.freeze({ width: 0, height: 0 }) // highly dubious
 const MAX_SCALE = 50
 
-/**
- * Generate array of coloured positions (tiles)
- */
-function generateTiles() {
-  // Mock tiles
-  const small = [
-    { x: 0, y: 0, colour: 'red' },
-    { x: 0, y: 5, colour: 'yellow' },
-    { x: 0, y: 2, colour: 'cyan' },
-    { x: 10, y: 0, colour: 'blue' },
-    { x: 10, y: 50, colour: 'green' },
-    { x: 4, y: 20, colour: 'white' },
-  ]
-  const tiles = []
-  let index = 0
-  for (let x = 0; x < mapSize.width; x++) {
-    for (let y = 0; y < mapSize.height; y++) {
-      const randomIndex = Math.floor(Math.random() * colours.length)
-      tiles.push({ x, y, colour: colours[randomIndex] })
-      /*tiles.push({ x, y, colour: small[index].colour })
-      index = (index + 1) % small.length*/
-    }
-  }
-  return tiles
-}
-
-export default function Map({ setPosition, cursorColour }) {
+export default function Map({
+  setPosition,
+  cursorColour
+}: {
+  setPosition: (position: Position) => void
+  cursorColour?: Colour | `url('/cursor.svg')`
+}) {
   //TODO diff datastructure
-  const [tiles, setTiles] = useState(generateTiles)
+  const [tiles, setTiles] = useState(generateTiles(mapSize))
```

Comment:
Change this to a lambda so it's only evaluated when needed
```suggestion
  const [tiles, setTiles] = useState(() => generateTiles(mapSize))
```

Reply:
Oh sick